queries.sql

```diff
@@ -21,3 +21,74 @@ WHERE name != 'Gabumon';
 
 SELECT * FROM animals
 WHERE weight_kg between 10.4 and 17.3;
+
+-- changing the species 
+
+update animals
+set species = 'digimon'
+where name like '%mon';
+
+update animals
+set species = 'pokemon'
+where species is null;
+
+-- deleting animals and rollback
+
+begin;
+
+delete from animals;
+
+rollback;
+
+-- deleting animals with a specific date of birth + using savepoints-- 
+
+begin;
+
+delete from animals
+where date_of_birth > '2022-01-01';
+
+savepoint savepoins_1;
+
+update animals
+set weight_kg = weight_kg * -1;
+
+rollback to savepoins_1;
+
+-- updating animals weight -- 
+
+update animals
+set weight_kg = weight_kg * -1
+where weight_kg < 1;
+
+commit;
+
+-- How many animals are there? --
+
+select count(id) from animals;
+
+-- How many animals have never tried to escape? --
+
+select count(escape_attempts) from animals
+where escape_attempts = '0';
+
+--What is the average weight of animals?
+
+select avg(weight_kg) from animals;
+
+--Who escapes the most, neutered or not neutered animals? --
+
+select count(escape_attempts) from animals
+group by neutered;
+
```

Comment:
- [ ] Ouch! this query doesn't necessarily show whether it's the "neutered" or "non-neutered" that escapes the most you might need to add the `neutered` column or another adjustment.